### PR TITLE
Added api reference hyperlinks in the Getting Started section [8454]

### PR DESCRIPTION
**Pull request by rsanchez15**
This pull request adds hyperlinks to the API Reference and fixes errors in the code snippets of the tutorial in the Getting Started section.

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/362/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/378/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/381/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/382/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/387/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/390/)

File: docs/fastrtpsgen/pubsub_app/includes/first_app.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Generating a minimal functional example
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The FASTRTPSGEN tool is used to build the publication/subscription application.
+The Fast RTPS-Gen tool is used to build the publication/subscription application.
```
Comment (EduPonz) [inline docs/fastrtpsgen/pubsub_app/includes/first_app.rst:4]:
```suggestion
The Fast DDS-Gen tool is used to build the publication/subscription application.
```

File: docs/fastrtpsgen/pubsub_app/includes/intro.rst [hunk 2]
```diff
@@ -24,7 +24,7 @@ presented in :ref:`fastrtpsgen_intro`.
     };
 
 Now we need to translate this file to something Fast RTPS understands.
-For this we have a code generation tool called FASTRTPSGEN (see :ref:`fastrtpsgen_use`), which can do two different
+For this we have a code generation tool called Fast RTPS-Gen (see :ref:`fastrtpsgen_use`), which can do two different
```
Comment (EduPonz) [inline docs/fastrtpsgen/pubsub_app/includes/intro.rst:27]:
```suggestion
For this we have a code generation tool called Fast RTPS-Gen (see :ref:`fastrtpsgen_use`), which can do two different
```
```suggestion
For this we have a code generation tool called Fast DDS-Gen (see :ref:`fastrtpsgen_use`), which can do two different
```

File: docs/spelling_wordlist.txt [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+Intra
+Nagle
+PKI
+QoS
+Structs
+TCP
+Topologies
+TypeObject
+WString
+XTypes
```
Comment (EduPonz) [inline docs/spelling_wordlist.txt:23]: Watch put with this. Probably they are already there and it's just a by-product of the rebase. You can try to sort alphabetically to check

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/394/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/396/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/397/)

**Pull request merged**